app/scripts/modules/netflix/pipeline/stage/canary/canaryStage.js

```diff
@@ -33,6 +35,84 @@ module.exports = angular.module('spinnaker.netflix.pipeline.stage.canaryStage',
             stageTypes: ['bake', 'findAmi', 'findImage', 'findImageFromTags'],
             message: 'You must have a Bake or Find AMI stage before a canary stage.'
           },
+          {type: 'requiredField', fieldName: 'canary.canaryConfig.lifetimeHours', fieldLabel: 'Canary Lifetime'},
+          {type: 'requiredField', fieldName: 'baseline.account', fieldLabel: 'Account'},
+          {type: 'requiredField', fieldName: 'baseline.cluster', fieldLabel: 'Cluster'},
+          {type: 'requiredField', fieldName: 'clusterPairs', fieldLabel: 'Cluster Pairs'},
+          {type: 'requiredField', fieldName: 'canary.canaryConfig.canaryAnalysisConfig.name', fieldLabel: 'Configuration'},
+          {type: 'custom', fieldLabel: 'Lookback Duration', validate: (_pipeline, stage) => {
+            const cac = stage.canary.canaryConfig.canaryAnalysisConfig,
+              useLookback = cac.useLookback,
+              lookbackMins = cac.lookbackMins;
+            let result = null;
+            if (useLookback && (!lookbackMins || lookbackMins === 0)) {
+              result = 'When an analysis type of <strong>Sliding Lookback</strong> is selected, the lookback duration must be positive.';
+            }
+
+            return result;
+          }},
+          {type: 'custom', fieldLabel: 'Report Frequency', validate: (_pipeline, stage) => {
+            const reportFrequency = stage.canary.canaryConfig.canaryAnalysisConfig.canaryAnalysisIntervalMins;
+            let result = null;
+            if (!reportFrequency || (reportFrequency < 1)) {
+              result = 'The <strong>Report Frequency</strong> is required and must be positive.';
```

Comment:
0 > x > 1 would fail the test, but would satisfy the message. Should either check for `!(reportFrequency > 0)` or fix message to be clear.